clash-lib/src/Clash/Normalize/Transformations/DEC.hs

```diff
@@ -27,79 +25,217 @@
 -}
 
 {-# LANGUAGE CPP #-}
+{-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE RecursiveDo #-}
 {-# LANGUAGE TemplateHaskell #-}
 
-module Clash.Normalize.DEC
-  (collectGlobals
-  ,collectGlobalsArgs
-  ,isDisjoint
-  ,mkDisjointGroup
-  )
-where
-
--- external
-import           Control.Concurrent.Supply        (splitSupply)
-import qualified Control.Lens                     as Lens
-import           Data.Bifunctor                   (second)
-import           Data.Bits                        ((.&.),complement)
-import           Data.Coerce                      (coerce)
-import qualified Data.Either                      as Either
-import qualified Data.Foldable                    as Foldable
-import qualified Data.Graph                       as Graph
-import qualified Data.IntMap.Strict               as IM
-import qualified Data.IntSet                      as IntSet
-import qualified Data.List                        as List
-import qualified Data.List.Extra                  as List
-import qualified Data.Map.Strict                  as Map
-import qualified Data.Maybe                       as Maybe
-import           Data.Monoid                      (All (..))
+module Clash.Normalize.Transformations.DEC
+  ( disjointExpressionConsolidation
+  ) where
+
+import Control.Concurrent.Supply (splitSupply)
+import Control.Lens ((^.))
+import qualified Control.Lens as Lens
+import qualified Control.Monad as Monad
+import Data.Bifunctor (second)
+import Data.Bits ((.&.),complement)
+import Data.Coerce (coerce)
+import qualified Data.Either as Either
+import qualified Data.Foldable as Foldable
+import qualified Data.Graph as Graph
+import Data.IntMap.Strict (IntMap)
+import qualified Data.IntMap.Strict as IntMap
+import qualified Data.IntSet as IntSet
+import qualified Data.List as List
+import qualified Data.List.Extra as List
+import qualified Data.Map.Strict as Map
+import qualified Data.Maybe as Maybe
+import Data.Monoid (All (..))
+import qualified Data.Text as Text
+import GHC.Stack (HasCallStack)
 
 #if MIN_VERSION_ghc(8,10,0)
-import           GHC.Hs.Utils                     (chunkify,mkChunkified)
+import GHC.Hs.Utils (chunkify,mkChunkified)
 #else
-import           HsUtils                          (chunkify,mkChunkified)
+import HsUtils (chunkify,mkChunkified)
```

Comment:
Commenting because you're changing it for style anyway: could you insert spaces between the commas?